modules/tracker/controllers/components/ApisubmissionComponent.php

```diff
@@ -152,29 +160,64 @@ public function post($args)
         $submissionModel = MidasLoader::loadModel('Submission',
                                                   $this->moduleName);
 
+        /** @var Tracker_ProducerModel $producerModel */
+        $producerModel = MidasLoader::loadModel('Producer', $this->moduleName);
+
         if (!isset($args['uuid'])) {
             /** @var UuidComponent $uuidComponent */
             $uuidComponent = MidasLoader::loadComponent('UuidComponent');
             $args['uuid'] = $uuidComponent->generate();
         }
 
+        if (isset($args['extraUrls'])) {
+            $args['extra_urls'] = json_decode($args['extraUrls'], true);
+        }
+
+        $args['build_results_url'] = isset($args['buildResultsUrl']) ? $args['buildResultsUrl'] : '';
+        $args['branch'] = isset($args['branch']) ? $args['branch'] : '';
+        $args['reproduction_command'] = isset($args['reproductionCommand']) ? $args['reproductionCommand'] : '';
+
+        $submitTime = strtotime($args['submitTime']);
+        if ($submitTime === false) {
+            throw new Exception('Invalid submitTime value: '.$args['submitTime'], -1);
+        }
+        $submitTime = date('Y-m-d H:i:s', $submitTime);
+        $args['submit_time'] = $submitTime;
+        $args['producer_revision'] = trim($args['producerRevision']);
+        $args['producer_id'] = $producerModel->getByCommunityIdAndName(
+            $args['communityId'], $args['producerDisplayName'])->getKey();
+
+        // Remove params from the submission args for later insertion in param table
+        if (isset($args['params']) && !is_null($args['params'])) {
+            $params = $args['params'];
+            unset($args['params']);
+        }
+
         /** @var Tracker_SubmissionDao $submissionDao */
         $submissionDao = $submissionModel->initDao('Submission',
```

Comment:
if we call createSubmission on the submission model it will create all the parameters automatically and we can remove the below block that deals with params here.